server/app/routes/v1/grids/grid_certificates.rb

```diff
@@ -1,11 +1,18 @@
-
 V1::GridsApi.route('grid_certificates') do |r|
 
   # POST /v1/grids/:name/certificates
   r.post do
     data = parse_json_body
     data[:grid] = @grid
-    outcome = GridCertificates::RequestCertificate.run(data)
+
+    if data['domains']
+      outcome = GridCertificates::RequestCertificate.run(data)
+    elsif data['certificate']
+      outcome = GridCertificates::Import.run(data)
```

Comment:
Needs updates to API docs... and maybe these should be separate API endpoints rather than having two different JSON request payloads.

Reply:
@jakolehm @jnummelin opinions? There's now two separate `POST /v1/grids/:grid/certificates` APIs with different JSON payloads: 

* https://github.com/kontena/kontena/blob/9a976607ac0a7ca28aa89ec2b16fb85c1dbe39f9/server/docs/source/index.html.md#request-lets-encrypt-certificate
* https://github.com/kontena/kontena/blob/9a976607ac0a7ca28aa89ec2b16fb85c1dbe39f9/server/docs/source/index.html.md#import-certificate

Reply:
IMO Two different payloads for same `POST` API sounds bad. Why not use `PUT /v1/certificates/my-grid/example.com` to import the cert?

Reply:
I agree with @jnummelin .

Reply:
I can change this to use `PUT /v1/certificates/:grid/:subject`, but then that also means that the `kontena certificate import` CLI command also needs to know the cert subject beforehand... I figure the CLI would take the expected subject CN as an argument, and then the server `GridCertificates::Import` mutation needs to validate that the certificate subject actually matches the expected subject?

Reply:
hmm, why couldn't CLI look into the cert and pass the correct subject?

Reply:
I can also do that... then both the CLI and the server will need to have some code to load the cert, but that shouldn't be too bad.

Reply:
Changed to use `PUT /v1/certificates/:grid/:subject` for the import. The server validates that the imported cert matches the subject. The CLI has an optional `--subject` param that defaults to the loaded cert's subject.